modules/cassandra/src/main/java/org/testcontainers/containers/CassandraContainer.java

```diff
@@ -0,0 +1,149 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.apache.commons.io.IOUtils;
+import org.testcontainers.containers.delegate.CassandraDatabaseDelegate;
+import org.testcontainers.delegate.DatabaseDelegate;
+import org.testcontainers.ext.ScriptUtils;
+import org.testcontainers.ext.ScriptUtils.ScriptLoadException;
+import org.testcontainers.utility.MountableFile;
+
+import javax.script.ScriptException;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+/**
+ * Cassandra container
+ * <p>
+ * Supports 2.x and 3.x Cassandra versions
+ *
+ * @author Eugeny Karpov
+ */
+public class CassandraContainer<SELF extends CassandraContainer<SELF>> extends GenericContainer<SELF> {
+
+    public static final String IMAGE = "cassandra";
+    public static final Integer CQL_PORT = 9042;
+    private static final String CONTAINER_CONFIG_LOCATION = "/etc/cassandra";
+    private static final String USERNAME = "cassandra";
+    private static final String PASSWORD = "cassandra";
+
+    private String configLocation;
+    private String initScriptPath;
+
+    public CassandraContainer() {
+        this(IMAGE + ":latest");
+    }
+
+    public CassandraContainer(String dockerImageName) {
+        this(dockerImageName, 3);
+    }
+
+    public CassandraContainer(String dockerImageName, int startupAttempts) {
+        super(dockerImageName);
+        setStartupAttempts(startupAttempts);
+    }
+
+    @Override
+    protected void configure() {
+        optionallyMapResourceParameterAsVolume(CONTAINER_CONFIG_LOCATION, configLocation);
+        addExposedPort(CQL_PORT);
+    }
+
+    @Override
+    protected void containerIsStarted(InspectContainerResponse containerInfo) {
+        runInitScriptIfRequired();
+    }
+
+    /**
+     * Load init script content and apply it to the database if initScriptPath is set
+     */
+    private void runInitScriptIfRequired() {
+        if (initScriptPath != null) {
+            try {
+                URL resource = Optional.ofNullable(Thread.currentThread().getContextClassLoader().getResource(initScriptPath))
```

Comment:
I believe that Optional is just better (forreadability) than ancient simple null check. Replaced by null check.
Also now this try block is similar to block in ContainerDatabaseDriver and can be extracted to ScriptUtils (since it's not bound to concrete database implementation).